packages/cactus-test-tooling/src/main/typescript/quorum/quorum-mp-test-ledger.ts

```diff
@@ -0,0 +1,255 @@
+import { EventEmitter } from "events";
+import Docker, { Container, ContainerCreateOptions } from "dockerode";
+import {
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+} from "@hyperledger/cactus-common";
+import { ITestLedger } from "../i-test-ledger";
+import { Containers } from "../common/containers";
+
+export interface IQuorumMultiPartyTestLedgerOptions {
+  readonly containerImageName?: string;
+  readonly containerImageVersion?: string;
+  readonly logLevel?: LogLevelDesc;
+  readonly emitContainerLogs?: boolean;
+  readonly envVars?: string[];
+  // For test development, attach to ledger that is already running, don't spin up new one
+  readonly useRunningLedger?: boolean;
+}
+
+export class QuorumMultiPartyTestLedger implements ITestLedger {
+  public readonly containerImageName: string;
+  public readonly containerImageVersion: string;
+  private readonly logLevel: LogLevelDesc;
+  private readonly emitContainerLogs: boolean;
+  private readonly useRunningLedger: boolean;
+  private readonly envVars: string[];
+
+  private readonly log: Logger;
+  public container: Container | undefined;
+  public containerId: string | undefined;
+
+  constructor(public readonly options?: IQuorumMultiPartyTestLedgerOptions) {
+    // @todo Replace with hyperledger ghcr link when available
+    this.containerImageName =
+      options?.containerImageName ||
+      "ghcr.io/outsh/cactus-quorum-multi-party-all-in-one";
+    this.containerImageVersion =
+      options?.containerImageVersion || "2022-03-14-65afd20";
+    this.logLevel = options?.logLevel || "info";
+
+    if (options?.emitContainerLogs != undefined) {
+      this.emitContainerLogs = options.emitContainerLogs;
+    } else {
+      this.emitContainerLogs = true;
+    }
+
+    if (options?.useRunningLedger != undefined) {
```

Comment:
@outSH Relying here only on the `!= undefined` check will let `this.emitContainerLogs` take up fun values like `{}` or `[]` or `0` or `"0"` (and many more, yaay Javascript...)
I recommend leveraging the strict boolean check from the `Bools` class similar to how it's done in the API server (and other places)
https://github.com/hyperledger/cactus/blob/release-v1.0.0/packages/cactus-cmd-api-server/src/main/typescript/api-server.ts#L113

The above will ensure that at least if garbage values come in as the constructor arg, we still default to either true or false (depending on what you want).
Another option would be (which is preferred if you don't mind making your parameter mandatory) to do an assertion with the `Checks` class that throws if the input parameter is anything but a `true` or a `false` with something like `Checks.truthy(Bools.isBooleanStrict(options.emitContainerLogs))` but this only works if all the parameters are mandatory so that's why the other places in the code mostly don't use this.